htdocs/core/modules/barcode/mod_barcode_product_standard.php

```diff
@@ -171,7 +206,23 @@ public function getNextValue($objproduct = null, $type = '')
 		$now = dol_now();
 
 		$numFinal = get_next_value($db, $mask, 'product', $field, $where, '', $now);
-
+		//if EAN13 calculate and substitute the last 13th character (* or ?) used in the mask by the EAN13 key
+		$literaltype = '';
+		$literaltype = $this->literalBarcodeType($db, $type);//get literal_Barcode_Type
+		if ($literaltype=='EAN13') //EAN13 rowid = 2
```

Comment:
The call $this->literalBarcodeType and test on == 'EAN13) should be done after the 
if ((substr($numFinal, -1)=='*') or (substr($numFinal, -1)=='?')
So we make a sql request only if there is the ? at end of string. This will save a sql when type is not ean13

Reply:
ok with @eldy
I sent an other code proposal:
Avoiding  sql call if no joker at the end of mask
Treating EAN13 in a more evolutive code form to prepare needs for other barcodes with key (if any).